glustercli/cmd/georep.go

```diff
@@ -346,8 +359,80 @@ func getRemoteClient(host string) (string, *restclient.Client, error) {
 	} else {
 		clienturl = fmt.Sprintf("%s://%s:%d", geoRepHTTPScheme, host, geoRepGlusterdPort)
 	}
-	client, err := restclient.New(clienturl, "", "", "", true)
-	return clienturl, client, err
+
+	remoteSecret := ""
+	// Secret is taken in following order of precedence (highest to lowest):
+	// --remote-secret
+	// --remote-secret-file
+	// GD2_REMOTE_AUTH_SECRET (environment variable)
+	// --secret
+	// --secret-file
+	// GD2_AUTH_SECRET (environment variable)
+	// --secret-file (default path)
+
+	// Remote Cluster secret --remote-secret
+	if flagRemoteSecret != "" {
+		remoteSecret = flagRemoteSecret
+	}
+
+	// Remote Cluster's secret file --remote-secret-file
+	if flagRemoteSecretFile != "" && remoteSecret == "" {
+		data, err := ioutil.ReadFile(flagRemoteSecretFile)
+		if err != nil {
+			failure(fmt.Sprintf("failed to read remote secret file %s", flagRemoteSecretFile),
+				err, 1)
+		}
+		remoteSecret = string(data)
+	}
+
+	// GD2_REMOTE_AUTH_SECRET
+	if remoteSecret == "" {
+		remoteSecret = os.Getenv("GD2_REMOTE_AUTH_SECRET")
+	}
+
+	// Below options of local cluster is used because --remote-* options
+	// are not specified and Remote volume may exists in same cluster where
+	// Master Volume exists
+
+	// Local custer --secret
+	if remoteSecret == "" && GlobalFlag.Secret != "" {
+		remoteSecret = GlobalFlag.Secret
+	}
+
+	// Local Cluster --secret-file
+	if GlobalFlag.SecretFile != "" && remoteSecret == "" {
+		data, err := ioutil.ReadFile(GlobalFlag.SecretFile)
```

Comment:
sorry, earlier I didn't think about this :(
currently, we have a function to `SetSecret`.  can we move the logic to new function to `getSecret` so that we can avoid code duplication here, as this also requires getting secret from the local cluster?

Reply:
I missed the enhancements done to CLI after sending this PR. Removed the duplicate code as the local secret is already set during `GlobalFlag.Init`